dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -499,15 +498,15 @@ public Tool registerManual(@ApiParam(hidden = true) @Auth User user,
         if (tool.isPrivateAccess() && Strings.isNullOrEmpty(tool.getToolMaintainerEmail())) {
             throw new CustomWebApplicationException("Tool maintainer email is required for private tools.", HttpStatus.SC_BAD_REQUEST);
         }
-        boolean releaseCreated = false;
         // populate user in tool
         tool.addUser(user);
         // create dependent Tags before creating tool
         Set<Tag> createdTags = new HashSet<>();
         for (Tag tag : tool.getWorkflowVersions()) {
             final long l = tagDAO.create(tag);
-            createdTags.add(tagDAO.findById(l));
-            releaseCreated = true;
+            Tag byId = tagDAO.findById(l);
+            createdTags.add(byId);
+            this.eventDAO.createAddTagToEntryEvent(user, tool, tag);
```

Comment:
Should the 3rd param by `byId` instead of `tag`? Don't know Hibernate well enough to say if it matters.

In DockerRepoTagResource, you use `byId`.

Reply:
For events, it looks like we add the full object.